File: native/windows/src/ScreenFix.Core/Configuration/DefaultConfiguration.cs
using ScreenFix.Core.Geometry;

namespace ScreenFix.Core.Configuration;

public static class DefaultConfiguration
{
    public static ScreenFixConfig Create(DisplayIdentity display)
    {
        ArgumentNullException.ThrowIfNull(display);

        var left = 1215d / 3440d;
        var width = (1920d - 1215d) / 3440d;
        RectD[] bands =
        [
            new(left, 0, width, 0.34),
            new(left, 0.34, width, 0.39),
            new(left, 0.73, width, 0.27),
        ];

        return new ScreenFixConfig(1, true, display, bands);
    }
}

File: native/macos/Sources/ScreenFixCore/DefaultConfiguration.swift
public enum DefaultConfiguration {
    public static func make(
        for display: DisplayIdentity,
        enabled: Bool
    ) -> ScreenFixConfiguration {
        let left = 1215.0 / 3440.0
        let width = (1920.0 - 1215.0) / 3440.0
        return ScreenFixConfiguration(
            schemaVersion: 1,
            enabled: enabled,
            display: display,
            bands: [
                NormalizedRect(x: left, y: 0.0, w: width, h: 0.34),
                NormalizedRect(x: left, y: 0.34, w: width, h: 0.39),
                NormalizedRect(x: left, y: 0.73, w: width, h: 0.27),
            ]
        )
    }
}
